Basic/Invaders/Assets/Scripts/EnemyBullet.cs

```diff
@@ -16,55 +16,74 @@ public class EnemyBullet : MonoBehaviour
     [SerializeField]
     ParticleSystem m_ShieldExplosionParticle;
 
+    void Awake()
+    {
+        enabled = false;
+    }
+
+    public override void OnNetworkSpawn()
+    {
+        enabled = IsServer;
+    }
+
     private void Start()
     {
         Assert.IsTrue(InvadersGame.Singleton);
-        Assert.IsTrue(NetworkManager.Singleton);
 
         if(InvadersGame.Singleton)
             InvadersGame.Singleton.isGameOver.OnValueChanged += OnGameOver;
     }
 
     private void Update()
     {
-        if (!NetworkManager.Singleton.IsServer) return;
+        if (!IsServer) return;
         transform.Translate(0, -m_TravelSpeed * Time.deltaTime, 0);
 
         if (transform.position.y < k_YBoundary) Destroy(gameObject);
     }
 
-    private void OnDestroy()
+    public override void OnDestroy()
     {
+        base.OnDestroy();
         if (InvadersGame.Singleton) InvadersGame.Singleton.isGameOver.OnValueChanged -= OnGameOver;
```

Comment:
Now that this is a NetworkBehaviour, should this be moved to OnDespawn? And the subscription to OnSpawn instead of Start?